lib/hls/hls_parser.js

```diff
@@ -2287,6 +2342,13 @@ shaka.hls.HlsParser = class {
     }
 
     if (mimeType == 'video/mp4' || mimeType == 'audio/mp4') {
+      if (clearKey) {
+        shaka.log.alwaysWarn('AES-128 in MP4 HLS is not yet supported.');
+        throw new shaka.util.Error(
+            shaka.util.Error.Severity.CRITICAL,
+            shaka.util.Error.Category.MANIFEST,
+            shaka.util.Error.Code.HLS_AES_128_ENCRYPTION_NOT_SUPPORTED);
```

Comment:
Perhaps it would be good to add an extra parameter that says the mimetype is not supported.

```
throw new shaka.util.Error(
    shaka.util.Error.Severity.CRITICAL,
    shaka.util.Error.Category.MANIFEST,
    shaka.util.Error.Code.HLS_AES_128_ENCRYPTION_NOT_SUPPORTED,
    mimeType);
```

Reply:
I changed to the same error handling as above for raw and video/webm.